deploy-heroku.py

```diff
@@ -20,41 +27,51 @@ def get_latest_heroku_release_version(app: str, api_key: str) -> int:
     response.raise_for_status()
 
     latest_release, *_ = response.json()
-    return latest_release["version"]
+
+    return HerokuRelease(
+        id=latest_release["id"],
+        version=latest_release["version"],
+        status=latest_release["status"],
+    )
```

Comment:
Ich weiss nicht was sonst in `latest_release` drin ist, aber falls nur die 3 Sachen, man könnte das machen - ein bisschen weniger ätzend als jedes Zeug kopieren:
```suggestion
    return HerokuRelease(**latest_release)
```
ansonsten

```python
return HerokuRelease(**{key: value for key, value in latest_release.items() if key not in ("blacklist1", "blacklist2")})
```

Reply:
@zyv 
Was hälst du hiervon?

```python
return HerokuRelease.from_dict(latest_release)
```
Um es sauber zu machen, müssten wir das Image hier in einer Action bauen, das auf DockerHub pushen und dann in der Action immer das fertige Image nutzen.
Derzeit wird das zur laufzeit gemacht. Das geht aber leider nicht mehr, wenn wir Poetry nutzen wollen mit den Dependencies, da das sonst zu lange dauert...

Reply:
Wo hast du das denn gefunden? Das funktioniert für mich nicht. Und was ist an `return HerokuRelease(**latest_release)` unsauber??? Das ist Standard-Python dict expansion.

```python
import dataclasses

@dataclasses.dataclass
class Test:
    pass

Test.from_dict({})
```

```
Traceback (most recent call last):
  File "...", line 7, in <module>
    Test.from_dict({})
AttributeError: type object 'Test' has no attribute 'from_dict'
```

Reply:
https://pypi.org/project/dataclasses-json/

Das wird auch im Backend schon an manchen Stellen genutzt.
Keys die in der Response vorhanden sind, aber nicht in der Dataclass, werden ignoriert und es kommt nicht zu einem Fehler.

Reply:
```python
from dataclasses import dataclass
from dataclasses_json import dataclass_json


@dataclass_json
@dataclass
class HerokuRelease:
    id: str
    version: int
    status: str
```

Reply:
Alles klar - du meintest ob es ok ist eine externe Bibliothek zu nehmen und darauf beziehen - mir ist es egal.

Meine Frage war eher ob da tatsächlich andere keys gibt oder nicht - falls nein, könnte man das tun, was ich vorgeschlagen habe, falls nicht - muss man es so machen, wie du es gemacht hast.

Ich war erstmal dafür es mit minimale Äderungen hinzukriegen, aber so passt es für mich auch.